go/subtle/util.go

```diff
@@ -85,11 +88,18 @@ func GetCurve(curve string) elliptic.Curve {
 }
 
 // ComputeHash calculates a hash of the given data using the given hash function.
-func ComputeHash(hashFunc func() hash.Hash, data []byte) []byte {
+func ComputeHash(hashFunc func() hash.Hash, data []byte) (hashed []byte, err error) {
+	if hashFunc == nil {
+		err = errNilHashFunc
+		return
+	}
 	h := hashFunc()
-	h.Write(data)
-	ret := h.Sum(nil)
-	return ret
+	_, err = h.Write(data)
+	if err != nil {
+		return
```

Comment:
why don't you return the error here?

Reply:
Ah it seems that you already name err as the returned error.

I'm not sure why this style is better, but it's inconsistent with the rest of our library, so I'd prefer not doing it.

Reply:
Named return parameters often use in sort function, for self document https://golang.org/doc/effective_go.html#named-results

Anyway, it does not matter in this case.